### PR TITLE
GEOPY-126

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-126 - Unit test: filter_xy** 
 refactor and redesign of utils.filter_xy() with supporting implementations of downsample_xy, downsample_grid, and window_xy.  Tested.

### CODE REVIEW

**Review by domfournier — DISMISSED** [review]
Great job again. The split is really nice.

File: geoapps/utils/utils.py [hunk 1]
```diff
@@ -390,44 +401,169 @@ def filter_xy(
...
     else:
-        filter_xy = np.ones_like(x, dtype="bool")
-    return filter_xy * mask
+        msg = f"Missing window keys: 'center' and 'size'."
+        raise KeyError(msg)
+
+    window_mask = x > x_lim[0]
+    window_mask &= x < x_lim[1]
+    window_mask &= y > y_lim[0]
+    window_mask &= y < y_lim[1]
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoapps/utils/utils.py:411]: yep, agreed!

File: geoapps/utils/utils.py [hunk 2]
```diff
@@ -361,27 +362,37 @@ def weighted_average(
...
+    Window x, y coordinates with window limits built from center and size.
+
+    Notes
+    -----
 
-    :param x: Easting coordinates
-    :param y: Northing coordinates
-    :param distance: Minimum distance between neighbours
+    This formulation knows nothing about rotation, so 'rotated' coordinates
+    should first be 'un-rotated' to a north-south, east-west orientation.
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoapps/utils/utils.py]: Updated the docstring, let me know if it's still too out of left field.  I can just remove it altogether - I just thought I should distinguish from filter_xy behaviour.

**Review by domfournier — APPROVED** [review]
Great!!

**Pull request merged**